list_books.php

```diff
@@ -0,0 +1,34 @@
+<?php
+include 'database.php';
+
+$query = "SELECT * FROM books";
```

Comment:
The current implementation lacks unit tests for listing books. While this functionality is straightforward, consider adding tests to ensure that the expected data is retrieved from the database.

```php
use PHPUnit\Framework\TestCase;

class BookRepositoryTest extends TestCase {
    public function testListBooks() {
        $mockConnection = $this->createMock(mysqli::class);
        $mockResult = $this->createMock(mysqli_result::class);

        $mockConnection->method('query')->willReturn($mockResult);
        $mockResult->method('fetch_assoc')->willReturnOnConsecutiveCalls(
            ['title' => 'Book 1', 'author' => 'Author 1', 'year' => 2020],
            null
        );

        $repository = new BookRepository($mockConnection);
        $books = $repository->listBooks();

        $this->assertEquals('Book 1', $books[0]['title']);
        $this->assertEquals('Author 1', $books[0]['author']);
        $this->assertEquals(2020, $books[0]['year']);
    }
}
```